core-customize/hybris/bin/custom/sapcxtools/sapcommercetoolkit/testsrc/tools/sapcx/commerce/toolkit/testing/testdoubles/user/UserServiceFake.java

```diff
@@ -92,17 +153,18 @@ public UserGroupModel getUserGroup(String s) {
 
 	@Override
 	public UserGroupModel getUserGroupForUID(String s) {
-		return null;
+		return userGroups.get(s);
 	}
 
 	@Override
 	public <T extends UserGroupModel> T getUserGroupForUID(String s, Class<T> aClass) {
-		return null;
+		UserGroupModel userGroup = getUserGroupForUID(s);
+		return (aClass.isInstance(userGroup)) ? aClass.cast(userGroup) : null;
 	}
 
 	@Override
 	public Set<UserGroupModel> getAllUserGroupsForUser(UserModel userModel) {
-		return null;
+		return new HashSet<>(userGroups.values());
```

Comment:
This is not expected. We should provide method "registerUserWithGroup" within the builder that will link the user and the group bidirectional and really return the "true" data. Otherwise, we couldnt use the fake for testing if some groups are linked and some are not.